test/beast/core/detect_ssl.cpp

```diff
@@ -166,9 +171,110 @@ class detect_ssl_test : public unit_test::suite
         }
     }
 
+#if BOOST_ASIO_HAS_CO_AWAIT
+    void testAsyncReadCoro()
```

Comment:
I think this test should just instantiate the initiating function with a `net::use_awaitable` (for example, by taking the address of it)

Reply:
pdimov  17:33
the test tests exactly what it needs - the actual "customer" scenario
17:34
no sane user just takes the address of the function